node-gui/src/backend_controller.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2021-2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common::chain::ChainConfig;
+use node_lib::node_controller::NodeController;
+use std::fmt::Debug;
+use std::sync::Arc;
+use tokio::sync::oneshot;
+
+pub struct NodeBackendController {
+    chain_config: Arc<ChainConfig>,
+    controller: NodeController,
+    manager_join_handle: Option<tokio::task::JoinHandle<()>>,
+}
+
+impl Debug for NodeBackendController {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("NodeInitializationData")
+            .field("chain_config", &self.chain_config)
+            .finish()
+    }
+}
+
+impl NodeBackendController {
+    pub async fn initialize() -> anyhow::Result<NodeBackendController> {
+        let (node_controller_sender, node_controller_receiver) = oneshot::channel();
+
+        if std::env::var("RUST_LOG").is_err() {
+            std::env::set_var("RUST_LOG", "info");
+        }
```

Comment:
Variable not set is not the only error condition here. It may be appropriate to handle the `NotUnicode` variant differently. (I am inclined to say a log entry would be nice but that is a chicken and egg problem)

https://doc.rust-lang.org/std/env/enum.VarError.html

Reply:
Yeah, I know. I saw the enum in there, but frankly I don't care. If someone setup a wrong value for the env var, I think it's OK to set it to a default value in the scope of the program.